homeassistant/components/hypontech/__init__.py

```diff
@@ -0,0 +1,63 @@
+"""The Hypontech Cloud integration."""
+
+from __future__ import annotations
+
+from hyponcloud import AuthenticationError, HyponCloud
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+from .coordinator import HypontechData, HypontechDataCoordinator
+
+_PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+type HypontechConfigEntry = ConfigEntry[HypontechData]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: HypontechConfigEntry) -> bool:
+    """Set up Hypontech Cloud from a config entry."""
+    session = async_get_clientsession(hass)
+    hypontech_cloud = HyponCloud(
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        session,
+    )
+    try:
+        if not await hypontech_cloud.connect():
+            raise ConfigEntryNotReady(
+                "Cannot connect to Hypontech Cloud. You may retry later."
+            )
+    except AuthenticationError as ex:
+        raise ConfigEntryAuthFailed("Authentication failed for Hypontech Cloud") from ex
+    except TimeoutError as ex:
+        raise ConfigEntryNotReady("Timeout connecting to Hypontech Cloud") from ex
```

Comment:
Your connect logic does not return False or None, so instead you should just call `await hypontech_cloud.connect()` without the conditional statement.

Related: https://github.com/jcisio/hyponcloud/blob/41c937ab8254c54c639be839e6a2914902cbd0ab/hyponcloud/client.py#L62

```python
    try:
        await hypontech_cloud.connect()
    except AuthenticationError as ex:
        raise ConfigEntryAuthFailed("Authentication failed for Hypontech Cloud") from ex
    except TimeoutError as ex:
        raise ConfigEntryNotReady("Timeout connecting to Hypontech Cloud") from ex
```

Reply:
`connect()` still returns false with unknown error (neither 401 nor 429).

Reply:
In this case you should raise an appropriate exception and catch that instead.